### ISSUE TITLE
Sample "TwoProjectMigrations" can't load assembly for migration

### ISSUE THREAD

**Issue by Slx-hub** [dotnet/EntityFramework.Docs#4840]
I was refactoring my application to support multiple DB providers in accordance with the "TwoProjectMigrations" sample, hitting a roadblock on actually using the migrations for migrating. Turns out I get the same error with the sample:

I hijacked the Worker.Execute() method to run a migration as follows:

```
using var scope = _services.CreateScope();
var services = scope.ServiceProvider;
var context = services.GetService<BlogContext>();
context.Database.Migrate();
```

Resulting in this error:

`System.IO.FileNotFoundException: Could not load file or assembly 'SqlServerMigrations, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.`

Is this not the right approach?

**Comment by roji** [comment]
Are you saying that the sample code as-is produces this error, without any changes?

**Comment by Slx-hub** [comment]
Thank you for the quick response!

Without modification it does run, but in my opinion the bare sample does not show complete functionality.
I guess I could have explained this more clearly, let me try again.

From my understanding the sample shows a successful project setup, but does not actually prove a working multi-provider migration.

An Example: The Worker.Execute() method from the sample invites to "do something interesting"

If I add a new Person for example:

```
db.People.Add(new Person());
db.SaveChanges();
```

I receive the error `no such table: People`

This is in line with my expectation, as the sample does configure the correct assembly for migrating: `x.MigrationsAssembly("SqlServerMigrations")`, 
but does not actually initiate a migration, which is the part that fails in my own project.

In the next step I added `db.Database.Migrate();` to test if the sample is unable to actually execute a migration as is the case with my project.

Turns out i receive the same exception: `System.IO.FileNotFoundException: Could not load file or assembly 'SqlServerMigrations, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.`

So now I am wondering if my approach was supposed to work or if I am misunderstanding how migrations should be executed in this scenario.

**Comment by roji** [comment]
Any chance you can post the modified sample so it's easier to know exactly what you're attempting? In general, code is much better than trying to describe things.

**Comment by Slx-hub** [comment]
This is essentially the diff of `Worker.Execute()` inside the TwoProjectMigrations sample:

```
    private void Execute()
    {
        using var scope = _services.CreateScope();
        var services = scope.ServiceProvider;

        var db = services.GetService<BlogContext>();

        // TODO: Something interesting

+       db.Database.Migrate();
+       db.People.Add(new Person());
+       db.SaveChanges();
    }
```

My question boils down to: should `db.Database.Migrate()` work in this sample or do migrations have to be applied differently?

Thank you for your continued support!

**Comment by Slx-hub** [comment]
Hey, were you able to have a look into my issue?

Many thanks!
